onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -41,14 +45,30 @@ std::vector<std::unique_ptr<RewriteRule>> GenerateRewriteRules(const Transformer
   return rules;
 }
 
-std::vector<TransformerProviderSet> GenerateTransformers(const TransformerLevel& level, 
+std::unique_ptr<RuleBasedGraphTransformer> GenerateRuleBasedGraphTransformer(const TransformerLevel& level,
+                                                                             const std::vector<std::string>* rules_to_enable,
+                                                                             std::string t_name) {
+  auto rewrite_rules_to_register = transformer_utils::GenerateRewriteRules(level, rules_to_enable);
+  if (rewrite_rules_to_register.empty()) {
+    return std::unique_ptr<RuleBasedGraphTransformer>{};
+  }
+
+  std::unique_ptr<RuleBasedGraphTransformer> rule_transformer =
+      std::make_unique<TopDownRuleBasedTransformer>(t_name + "_RuleBasedTransformer",
+                                                    "Apply rewrite rules for " + t_name);
+  for (auto& entry : rewrite_rules_to_register) {
+    rule_transformer->Register(std::move(entry));
+  }
+
+  return rule_transformer;
+}
+
+std::vector<TransformerProviderSet> GenerateTransformers(const TransformerLevel& level,
                                                          const std::vector<std::string>* transformers_to_enable) {
   std::vector<TransformerProviderSet> transformers;
   switch (level) {
-    case TransformerLevel::Level1: {
-      std::vector<std::string> l1_execution_providers = {};
-      transformers.emplace_back(std::make_unique<UnsqueezeElimination>(), l1_execution_providers);
-    } break;
+    case TransformerLevel::Level1:
+      break;
 
```

Comment:
Maybe a comment here saying there are currently no L1 transformers that aren't handled by GenerateRuleBasedGraphTransformer would be helpful

Reply:
Agreed. But I would rather add in there the generation of the rule-based transformer.
So when we generate the transformers, one of the generated ones will be the rule-based transformer.
I don't think there is need to expose the generation of rewrite rules (they cannot exist outside a rule-based transformer).
This will simplify the AddPredefinedTransformers in the inference_session too.